libraries/plugins/witness/witness.cpp

```diff
@@ -161,6 +163,21 @@ void witness_plugin::refresh_witness_key_cache()
    }
 }
 
+void witness_plugin::stop_production_loop()
+{
+   try {
+      if( _block_production_task.valid() )
+      {
+         fc::scoped_lock<boost::mutex> lock( _production_loop_mutex );
+         _block_production_task.cancel_and_wait(__FUNCTION__);
```

Comment:
This will result in a deadlock.
The lock is held until ``cancel_and_wait`` returns, but ``schedule_production_loop`` blocks on the same lock, so it will never complete and ``cancel_and_wait`` will never return.